reporter/src/main/kotlin/Reporter.kt

```diff
@@ -41,20 +41,26 @@ abstract class Reporter {
         val ALL by lazy { LOADER.iterator().asSequence().toList() }
     }
 
+    /**
+     * The default output filename to use with this reporter format.
+     */
+    abstract val defaultFilename: String
```

Comment:
"default" sounds like there would be a way to change it. What about `reportFilename`?

Reply:
That was exactly the intention, because there *is* a way to use another name by the caller, as the caller now decides where to write the `OutputStream` to. This is supposed to be the name that makes the most sense in the context of that format.

Initially, I only wanted to store the file extension here, similar to like we do in `OutputFormat`, but just storing the file extension is not enough as e.g. static and web-app reports would clash then (both use `.html`) and `NOTICE` simply has no extension at all.

Reply:
True, I missed that.